src/main/java/software/amazon/event/kafkaconnector/mapping/JsonPathDetailTypeMapper.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.mapping;
+
+import static com.jayway.jsonpath.Configuration.defaultConfiguration;
+import static com.jayway.jsonpath.Option.SUPPRESS_EXCEPTIONS;
+import static java.lang.String.format;
+import static java.util.Collections.singletonMap;
+import static org.apache.kafka.connect.json.JsonConverterConfig.SCHEMAS_ENABLE_CONFIG;
+import static software.amazon.event.kafkaconnector.EventBridgeSinkConfig.AWS_DETAIL_TYPES_MAPPER_JSON_PATH_MAPPER_FIELDREF;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import software.amazon.event.kafkaconnector.EventBridgeSinkConfig;
+import software.amazon.event.kafkaconnector.logging.ContextAwareLoggerFactory;
+
+public class JsonPathDetailTypeMapper implements DetailTypeMapper {
+  private static final Logger log =
+      ContextAwareLoggerFactory.getLogger(JsonPathDetailTypeMapper.class);
+
+  private static final Configuration jsonPathConfiguration =
+      defaultConfiguration()
+          .addOptions(
+              SUPPRESS_EXCEPTIONS // suppress exception otherwise
+              // com.jayway.jsonpath.ReadContext#read throws an exception if JSON path could
+              // not
+              // be found
+              );
+
+  private String jsonPath;
+  private final JsonConverter jsonConverter;
+  private final ObjectMapper objectMapper;
+
+  public JsonPathDetailTypeMapper() {
+    this.jsonConverter = new JsonConverter();
+    this.objectMapper = new ObjectMapper();
+    this.jsonConverter.configure(singletonMap(SCHEMAS_ENABLE_CONFIG, "false"), false);
+  }
+
+  @Override
+  public String getDetailType(SinkRecord record) {
+    if (record == null) {
```

Comment:
Can you think of any use case where this would be null? Afaik Connect would only deliver valid _SinkRecords that have been read from Kafka and include the kafkaOffset of the record in the Kafka topic-partition in addition to the standard fields (From the [docs](https://kafka.apache.org/20/javadoc/org/apache/kafka/connect/sink/SinkRecord.html#kafkaOffset))._

In addition, at that stage we have gone through several chains and stages before and should ideally filter null values there without making it the mapper responsibility to check. Wdyt?

Reply:
Yup, this is really just to make some linters/reviews happy - should never happen. But I don't think it hurts to have it in the code base explicitly.

Reply:
As far as I can remember it should be optimized/removed by the JVM HotSpot during runtime.